src/FieldDateTime.tsx

```diff
@@ -0,0 +1,316 @@
+      }
+      label={label}
+      maxDate={maxDate}
+      maxTime={maxTime}
+      minDate={minDate}
+      minTime={minTime}
+      onChange={generateOnChangeHandler({ name, setFieldValue, onChange })}
+      placeholderText={placeholderText}
+      popperPlacement={popperPlacement}
+      required={required}
+      selectLocalDateTime={selectLocalDateTime}
+      size={size}
+      timeFormat={timeFormat}
+      validationText={
+        (isTouched(touched, name) || submitCount > 0) && errors[name]
+          ? errors[name]
+          : null
+      }
+      value={value}
+      width={width}
+      withPortal={withPortal}
+    />
+  );
+
+/**
+ * @description This component serves as an adapter between the Formik Field component and
+ * the underlying pebble FieldDateTime component
+ *
+ * ref: https://pebble.istreamplanet.net/#/Components/FieldDateTime
+ */
+const FieldDateTimeAdapter = ({
+  autoFocus,
+  className,
+  dateFormat,
+  disabled,
+  excludeTime,
+  filterDate,
+  helpText,
+  hideLabel,
+  id,
+  isClearable,
+  label,
+  maxDate,
+  maxTime,
+  minDate,
+  minTime,
+  name,
+  onChange,
+  placeholderText,
+  popperPlacement,
+  required,
+  selectLocalDateTime,
+  size,
+  timeFormat,
+  type,
+  width,
+  withPortal,
+}: FieldDateTimeAdapterOptions): JSX.Element => (
+  <Field name={name} type={type}>
+    {generateFieldDateTimeRenderProp({
+      autoFocus,
+      className,
+      dateFormat,
+      disabled,
+      excludeTime,
+      filterDate,
+      helpText,
+      hideLabel,
+      id,
+      isClearable,
+      label,
+      maxDate,
+      maxTime,
+      minDate,
+      minTime,
+      onChange,
+      placeholderText,
+      popperPlacement,
+      required,
+      selectLocalDateTime,
+      size,
+      timeFormat,
+      width,
+      withPortal,
+    })}
+  </Field>
+);
+
+interface FieldDateTimeAdapterOptions extends InputOptions {
+  /**
+   * The id attribute of the input
+   */
+  id?: string;
+  /**
+   * Label for the input
+   */
+  label?: string;
+  /**
+   * Flag to automatically focus the input when true
+   */
+  autoFocus?: boolean;
+  /**
+   * Additional classNames to add
+   */
+  className?: string;
+  /**
+   * Format used to display text in the input box default is YYYY-MM-DD
+   * or YYYY-MM-DD {timeFormat} Z To support correct handling of
+   * inputting UTC times, the UTC offset should be included in the format
+   *
+   * ref: https://momentjs.com/docs/#/parsing/string-format/
+   */
+  dateFormat?: string;
+  /**
+   * Flag when true will make input disabled and non-focusable
+   */
+  disabled?: boolean;
+  /**
+   * Will disable time of day setting when true
+   */
+  excludeTime?: boolean;
+  /**
+   * Filter the selectable dates based on custom criteria
+   *
+   * @param {string} date - a UTC ISO 8601 string ref: https://en.wikipedia.org/wiki/ISO_8601
+   * @returns {boolean} - Flag that determines if the date is selectable by the user
```

Comment:
this should be also be defined in the Function type (e.g. it should have parameters types and a return type.